### ISSUE TITLE
Bug: v6's package.json types exports are incorrect

### ISSUE THREAD

**Issue by RebeccaStevens** [typescript-eslint/typescript-eslint#6803]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

utils

### Repro Code

```typescript
import { RuleTester } from "@typescript-eslint/utils/eslint-utils/rule-tester";
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "allowSyntheticDefaultImports": true,
    "allowUnreachableCode": false,
    "allowUnusedLabels": false,
    "alwaysStrict": true,
    "esModuleInterop": true,
    "exactOptionalPropertyTypes": true,
    "forceConsistentCasingInFileNames": true,
    "importHelpers": false,
    "lib": ["ESNext"],
    "module": "ESNext",
    "moduleResolution": "bundler",
    "newLine": "LF",
    "noEmitOnError": true,
    "noImplicitReturns": true,
    "noPropertyAccessFromIndexSignature": true,
    "noUncheckedIndexedAccess": true,
    "noUnusedLocals": false,
    "noUnusedParameters": false,
    "pretty": true,
    "resolveJsonModule": true,
    "sourceMap": false,
    "strict": true,
    "target": "ES2020"
  }
}
```

### Expected Result

Types to exists for the import.

### Actual Result

Types aren't imported.

### Additional Info

The issue is caused by an incorrect path in the "types" export.

```jsonc
"./eslint-utils/rule-tester": {
  "types": "./dist/eslint-utils/rule-tester/RuleTester.d.ts", // <-- File actually has .d.mts extension
  "default": "./dist/eslint-utils/rule-tester/RuleTester.js"
},
```

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `6.0.0-alpha.106` |
| `@typescript-eslint/parser`            | `6.0.0-alpha.106` |
| `@typescript-eslint/scope-manager`     | `6.0.0-alpha.106` |
| `@typescript-eslint/typescript-estree` | `6.0.0-alpha.106` |
| `@typescript-eslint/type-utils`        | `6.0.0-alpha.106` |
| `@typescript-eslint/utils`             | `6.0.0-alpha.106` |
| `TypeScript`                           | `5.0.2` |
| `ESLint`                               | `8.36.0` |
| `node`                                 | `18.15.0` |

**Comment by RebeccaStevens** [comment]
Ahhhhh, ooop. This is embarrassing. I accidently edited my node_modules folder when trying to run a batch command on all my project's .ts files.
